app/grants/views.py

```diff
@@ -258,10 +259,131 @@ def grants_by_grant_type(request, grant_type):
+            'is_on_team': is_grant_team_member(grant, request.user.profile) if request.user.is_authenticated else False,
+            'amount': grant_amount,
+            'clr_prediction_curve': grant.clr_prediction_curve,
+            'last_clr_calc_date': grant.last_clr_calc_date,
+            'safe_next_clr_calc_date': naturaltime(grant.safe_next_clr_calc_date) if grant.last_clr_calc_date else None,
+            'amount_received_in_round': grant.amount_received_in_round,
+            'positive_round_contributor_count': grant.positive_round_contributor_count,
+            'monthly_amount_subscribed': grant.monthly_amount_subscribed,
+            'is_clr_eligible': grant.is_clr_eligible
+        } for grant in grants],
+        'credentials': {
+            'is_staff': request.user.is_staff,
+            'is_authenticated': request.user.is_authenticated
+        }
+    })
+
+
+def build_grants_by_type(request, grant_type='', sort='weighted_shuffle', network='mainnet', keyword='', state='active', category='', following=False):
+    sort_by_clr_pledge_matching_amount = None
+    if 'match_pledge_amount_' in sort:
+        sort_by_clr_pledge_matching_amount = int(sort.split('amount_')[1])
+
+    _grants = Grant.objects.filter(
+        network=network, hidden=False
+    ).keyword(keyword)
+
+    _grants = _grants.order_by(sort, 'pk')
+    _grants.first()
+
+    if state == 'active':
+        _grants = _grants.active()
+
+    if grant_type != 'all':
+        _grants = _grants.filter(grant_type__name=grant_type)
+
+    if following and request.user.is_authenticated:
+        favorite_grants = Favorite.grants().filter(user=request.user).values('grant_id')
+        _grants = _grants.filter(id__in=Subquery(favorite_grants))
+
+    clr_prediction_curve_schema_map = {10**x: x+1 for x in range(0, 5)}
+    if sort_by_clr_pledge_matching_amount in clr_prediction_curve_schema_map.keys():
+        sort_by_index = clr_prediction_curve_schema_map.get(sort_by_clr_pledge_matching_amount, 0)
+        field_name = f'clr_prediction_curve__{sort_by_index}__2'
+        _grants = _grants.order_by(f"-{field_name}")
+
+    if category:
+        _grants = _grants.filter(Q(categories__category__icontains=category))
+
+    _grants = _grants.prefetch_related('categories')
+
+    for grant in _grants:
+        clr_round = None
+
+        if grant.in_active_clrs.count() > 0:
+            clr_round = grant.in_active_clrs.first()
+
+        if clr_round:
+            grant.is_clr_active = True
+            grant.clr_round_num = clr_round.round_num
+        else:
+            grant.is_clr_active = False
+            grant.clr_round_num = 'LAST'
+
+    return _grants
+
+
+def get_grant_types(network):
+    all_grants_count = 0
+    grant_types = []
+    for _grant_type in GrantType.objects.all():
+
+        count = Grant.objects.filter(
+            network=network, hidden=False, active=True, grant_type=_grant_type
+        ).count()
```

Comment:
^ could we make this call only once. I know this ported over code but 
We could do 

```
active_grants= Grant.objects.filter(
            network=network, hidden=False, active=True
 )

for _grant_type in GrantType.objects.all():
     count = active_grants.filter(grant_type=_grant_type).count()
     ...
```

Reply:
done